### PR TITLE
Use github.com/mailru/easyjson for JSON encoding and decoding of Targets and Results

### PR DESCRIPTION
**Pull request by tsenart**
It's much faster.

```
benchmark                            old ns/op     new ns/op     delta
BenchmarkEncodings/gob-encode-8      1278          1146          -10.33%
BenchmarkEncodings/gob-decode-8      131           128           -2.29%
BenchmarkEncodings/csv-encode-8      800           757           -5.38%
BenchmarkEncodings/csv-decode-8      705           657           -6.81%
BenchmarkEncodings/json-encode-8     2617          1212          -53.69%
BenchmarkEncodings/json-decode-8     6259          1547          -75.28%

benchmark                            old allocs     new allocs     delta
BenchmarkEncodings/gob-encode-8      2              2              +0.00%
BenchmarkEncodings/gob-decode-8      0              0              +0.00%
BenchmarkEncodings/csv-encode-8      2              2              +0.00%
BenchmarkEncodings/csv-decode-8      2              2              +0.00%
BenchmarkEncodings/json-encode-8     3              4              +33.33%
BenchmarkEncodings/json-decode-8     7              2              -71.43%

benchmark                            old bytes     new bytes     delta
BenchmarkEncodings/gob-encode-8      200           200           +0.00%
BenchmarkEncodings/gob-decode-8      0             0             +0.00%
BenchmarkEncodings/csv-encode-8      209           209           +0.00%
BenchmarkEncodings/csv-decode-8      192           192           +0.00%
BenchmarkEncodings/json-encode-8     801           1004          +25.34%
BenchmarkEncodings/json-decode-8     72            64            -11.11%
```

```
benchmark                                old ns/op     new ns/op     delta
BenchmarkJSONTargetEncoding/encode-8     787           781           -0.76%
BenchmarkJSONTargetEncoding/decode-8     5812          1510          -74.02%

benchmark                                old allocs     new allocs     delta
BenchmarkJSONTargetEncoding/encode-8     1              1              +0.00%
BenchmarkJSONTargetEncoding/decode-8     18             9              -50.00%

benchmark                                old bytes     new bytes     delta
BenchmarkJSONTargetEncoding/encode-8     470           470           +0.00%
BenchmarkJSONTargetEncoding/decode-8     1230          703           -42.85%
```

### CODE REVIEW

**Comment by tsenart** [comment]
TODO: Add dependency with dep.

File: lib/results_test.go [hunk 1]
```diff
@@ -103,3 +104,48 @@ func TestEncoding(t *testing.T) {
...
+		{"gob", NewEncoder, NewDecoder},
+		{"csv", NewCSVEncoder, NewCSVDecoder},
+		{"json", NewJSONEncoder, NewJSONDecoder},
+	} {
+		var buf bytes.Buffer
+		enc := tc.enc(&buf)
+
+		b.Run(tc.encoding+"-encode", func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				enc.Encode(&results[i%len(results)])
```
Comment (golangci) [inline lib/results_test.go]: Error return value of `enc.Encode` is not checked

File: lib/results_test.go [hunk 2]
```diff
@@ -103,3 +104,48 @@ func TestEncoding(t *testing.T) {
...
+			}
+		})
+
+		dec := tc.dec(&buf)
+		b.Run(tc.encoding+"-decode", func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				dec.Decode(&results[i%len(results)])
```
Comment (golangci) [inline lib/results_test.go]: Error return value of `dec.Decode` is not checked

File: lib/results_easyjson.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+			out.RawString(prefix[1:])
+		} else {
+			out.RawString(prefix)
+		}
+		out.String(string(in.Error))
+	}
+	{
+		const prefix string = ",\"body\":"
+		if first {
+			first = false
```
Comment (golangci) [inline lib/results_easyjson.go]: ineffectual assignment to `first`

File: lib/targets_test.go [hunk 4]
```diff
@@ -389,3 +389,34 @@ func TestErrNilTarget(t *testing.T) {
...
+			Header: http.Header{"Content-Type": []string{"high/energy"}},
+		}
+	}
+
+	var buf bytes.Buffer
+	enc := NewJSONTargetEncoder(&buf)
+
+	b.Run("encode", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			enc.Encode(&targets[i%len(targets)])
```
Comment (golangci) [inline lib/targets_test.go]: Error return value of `enc.Encode` is not checked

File: lib/targets_test.go [hunk 5]
```diff
@@ -389,3 +389,34 @@ func TestErrNilTarget(t *testing.T) {
...
+		}
+	})
+
+	dec := NewJSONTargeter(&buf, nil, nil)
+	b.Run("decode", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			dec.Decode(&targets[i%len(targets)])
```
Comment (golangci) [inline lib/targets_test.go]: Error return value of `dec.Decode` is not checked

File: lib/targets_easyjson.go [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+			out.RawString(prefix[1:])
+		} else {
+			out.RawString(prefix)
+		}
+		out.Base64Bytes(in.Body)
+	}
+	if len(in.Header) != 0 {
+		const prefix string = ",\"header\":"
+		if first {
+			first = false
```
Comment (golangci) [inline lib/targets_easyjson.go]: ineffectual assignment to `first`

File: lib/targets_easyjson.go [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+							if !in.IsDelim(']') {
+								v2 = make([]string, 0, 4)
+							} else {
+								v2 = []string{}
+							}
+						} else {
+							v2 = (v2)[:0]
+						}
+						for !in.IsDelim(']') {
+							var v3 string
```
Comment (golangci) [inline lib/targets_easyjson.go]: should merge variable declaration with assignment on next line

**Pull request merged**